Fix autosave/dialog flow

merapi opened the pull request:
## Summary

This PR will fix autosave triggered on every element update and 2 issues from [here](https://xwp.slack.com/archives/C013J2EGH8R/p1666172443879059).

There is one more [issue](https://github.com/GoogleForCreators/web-stories-wp/issues/12474#issuecomment-1281083460) with autosave but I need to think about it a little more, it can have a bigger impact on the app (`hasNewChanges` is used in a few places).

## Relevant Technical Choices

<!-- Please describe your changes. -->

## To-do

<!-- A list of things that need to be addressed in this PR or follow-up changes. -->

## User-facing changes

<!--
Please describe your changes.
Include before/after screenshots or a short video.
-->

## Testing Instructions

<!--
How can the changes in this PR be verified?
Please provide step-by-step instructions how to reproduce the issue, if applicable.
Write step-by-step test instructions aimed at non-tech-savvy users, even if the PR is not user-facing.
-->

<!-- ignore-task-list-start -->
- [ ] This is a non-user-facing change and requires no QA
<!-- ignore-task-list-end -->

This PR can be tested by following these steps:

1.


## Reviews

### Does this PR have a security-related impact?

<!-- Examples: new APIs, changes to KSES, etc.  -->

### Does this PR change what data or activity we track or use?

<!-- Examples: changes to telemetry, new third-party APIs -->

### Does this PR have a legal-related impact?

<!-- Examples: new images with unknown sources, new production dependencies with incompatible licenses -->

## Checklist

<!-- Check these after PR creation -->

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

<!--
Please reference the issue(s) this PR addresses.
No URLs, just the issue numbers.
Use "Fixes #123" if it fixes an issue.

NOTE: One reference per line!

Example:

Fixes #123
Partially addresses #456
See #789
-->

Fixes #12474

merapi committed:
Fix autosave/dialog flow

swissspidy reviewed (approved):
At first glance looks good 👍 

Dig the improved naming

merapi committed:
Merge branch 'main' into fix/12474-lock-autosave-issues

merapi committed:
First attempt

merapi committed:
Skip the autosave test for now

merapi committed:
Skip for now

swissspidy committed:
Get first post lock information directly on mount

- Preload first post lock API response so that it’s immediately available in the editor
- Immediately fetch information on mount, before interval is up
- Fix `act` warnings in tests

spacedmonkey commented on includes/templates/admin/edit-story.php:
```diff
@@ -88,6 +88,11 @@
+		[
+			'_embed' => 'author',
+		]
+	),
```
This was not preloaded by design. Why is it being preloaded now?

swissspidy replied:
See discussion at https://xwp.slack.com/archives/C01H2K83WD9/p1666788696201389 for an explanation.

The aim is to immediately get the current lock owner information right when opening the editor, without having to wait for the 60s interval to pass first. There's no harm in preloading the response and saving 1 HTTP request.

Note that preloading a response only stores it once. Any subsequent calls will trigger an HTTP request as expected. Just the first one is fetched from cache.

spacedmonkey reviewed (changes requested):
Seems like now, I am not seeing the message that says, the story has been taken over. Now, I see, when a story is taken over, I always see this message. 
<img width="632" alt="Screenshot 2022-10-26 at 16 22 23" src="https://user-images.githubusercontent.com/237508/198067666-9b3a4034-4574-4ce9-9a88-9300f6465dc1.png">

So this is how it should work. 

1. User 1, edits story. Sets lock. 
2. User 2, open story. 
3. User 2, takes over story. 
4. User 1, see message that story was taken over, without option to over story back over.

merapi committed:
Revert "Get first post lock information directly on mount"

This reverts commit 3e6f6c51ac1d4c5d61d8af52b06438bb178a39cb.
